### PR TITLE
fix Issue 14253 - [REG2.067b3] std.traits.ParameterStorageClassTuple giv...

### PR DESCRIPTION
**Pull request by WalterBright**
...es compiler errors when encountering 'return' functions

https://issues.dlang.org/show_bug.cgi?id=14253

Also requires: https://github.com/D-Programming-Language/dmd/pull/4486

### CODE REVIEW

**Comment by WalterBright** [comment]
added unittest

File: std/traits.d [hunk 1]
```diff
@@ -624,13 +625,15 @@ private template fqnType(T,
...
+            return format("%s%s%s%s%s%s%s%s",
                  attrs & FA.pure_ ? " pure" : "",
                  attrs & FA.nothrow_ ? " nothrow" : "",
                  attrs & FA.ref_ ? " ref" : "",
                  attrs & FA.property ? " @property" : "",
                  attrs & FA.trusted ? " @trusted" : "",
-                 attrs & FA.safe ? " @safe" : ""
+                 attrs & FA.safe ? " @safe" : "",
+                 attrs & FA.nogc ? " @nogc" : "",
+                 attrs & FA.return_ ? " return" : ""
```
Comment (monarchdodra) [inline std/traits.d]: Let's remember that D allows trailing commas, specifically for when you have a "list style" layout such as here. This is just a comment.

**Comment by monarchdodra** [comment]
Shouldn't the `ParameterStorageClass` enum also be updated?
https://github.com/WalterBright/phobos/blob/fix14253/std/traits.d#L947

Or is that not considered a storage class?

**Comment by WalterBright** [comment]
> Shouldn't the  ParameterStorageClass  enum also be updated?

yes, and done!

**Comment by WalterBright** [comment]
This is a regression fix - it needs to be pulled or an explanation of why not.

**Comment by monarchdodra** [comment]
LGTM. Could someone else take a look too? It's easy to miss something subtle in traits, and I wouldn't want to break anything by rush-merging this.

**Comment by WalterBright** [comment]
done

**Comment by MartinNowak** [comment]
Auto-merge toggled on

**Pull request merged**